dotnet/simple-app/Program.cs

```diff
@@ -0,0 +1,284 @@
+            // This means the history of changes has been tampered.
+            Console.WriteLine(e.ToString());
+            Environment.Exit(1);
+        }
+
+        string key1 = "key1", key2 = "key2";
+
+        await client.Set(key1, new byte[] { 1, 2, 3 });
+        await client.Set(key2, new byte[] { 4, 5, 6 });
+
+        List<string> keyList = new List<string>();
+        keyList.Add(key1);
+        keyList.Add(key2);
+
+        // A multi-key read.
+        List<Entry> result = await client.GetAll(keyList);
+
+        foreach (Entry e in result)
+        {
+            Console.WriteLine($"({string.Join(" ", key)}, {string.Join(" ", e.Value)})\n");
+        }
+
+        // History operations.
+        await client.Set("history", new byte[] { 1, 2, 3 });
+        await client.Set("history", new byte[] { 3, 2, 1 });
+
+        List<Entry> history = await client.History("history", 10, 0, false);
+
+        Console.WriteLine($"History of 'history', entry 1: ({Encoding.UTF8.GetString(history[0].Key)}, {string.Join(" ", history[0].Value)})\n");
+        Console.WriteLine($"History of 'history', entry 2: ({Encoding.UTF8.GetString(history[1].Key)}, {string.Join(" ", history[1].Value)})\n");
+
+        // Scan operations.
+        String prefix = "myKey";
+        key1 = prefix + "1";
+        key2 = prefix + "2";
+
+        await client.Set(key1, new byte[] { 1, 2, 3 });
+
+        await client.Set(key2, new byte[] { 4, 5, 6 });
+
+        // scan is usually done by a prefix.
+        // Of course, we can scan by a complete key name.
+        List<Entry> scan = await client.Scan(prefix, 2, false);
+
+        Console.WriteLine($"Scan results of '{prefix}', entry 1: ({Encoding.UTF8.GetString(scan[0].Key)}, {string.Join(" ", scan[0].Value)})\n");
+        Console.WriteLine($"Scan results of '{prefix}', entry 2: ({Encoding.UTF8.GetString(scan[1].Key)}, {string.Join(" ", scan[1].Value)})\n");
+
+        // zAdd, zScan operations.
+        String set = "mySet";
+        await client.ZAdd(set, key1, 2);
+        await client.ZAdd(set, key2, 2);
+
+        // Here, we do a zScan providing the `sinceTxId` which should be
+        // the latest transaction id we are interested in being considered.
+        List<ZEntry> zScan = await client.ZScan(set, 10, false);
+
+        Console.WriteLine($"Results of 'zScan', record 1: ({Encoding.UTF8.GetString(zScan[0].Key)},{string.Join(" ", zScan[0].Entry.Value)})\n");
+        Console.WriteLine($"Results of 'zScan', record 2: ({Encoding.UTF8.GetString(zScan[1].Key)},{string.Join(" ", zScan[1].Entry.Value)})\n");
+
+        await client.Close();
+    }
+
+    private static async Task SqlUsageExample()
+    {
+        var client = new ImmuClient(immudbServerAddress, 3322);
+        await client.Open("immudb", "immudb", "defaultdb");
+
+        await client.SQLExec("CREATE TABLE IF NOT EXISTS logs(id INTEGER AUTO_INCREMENT, created TIMESTAMP, entry VARCHAR, PRIMARY KEY id)");

```

Comment:
Can we split this and other SQL code into multi-line? Will be much easier to read:

```suggestion
        await client.SQLExec(@"
            CREATE TABLE IF NOT EXISTS logs(
                id INTEGER AUTO_INCREMENT,
                created TIMESTAMP,
                entry VARCHAR,
                PRIMARY KEY id
            )");
```

(although not sure if github will be able to display correctly, it messes up the diff view in this comment)

Reply:
done